migrate/v0_13/migrate.go

```diff
@@ -78,6 +89,211 @@ func MigrateCDP(oldGenState v0_11cdp.GenesisState) v0_13cdp.GenesisState {
+				v0_13hard.NewInterestRateModel(sdk.ZeroDec(), sdk.MustNewDecFromStr("0.002"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.01")),
+				sdk.MustNewDecFromStr("0.025"), sdk.MustNewDecFromStr("0.01"),
+			),
+			// busd
+			v0_13hard.NewMoneyMarket("busd", v0_13hard.NewBorrowLimit(true, sdk.MustNewDecFromStr("100000000000000"), sdk.MustNewDecFromStr("0.5")), "busd:usd", sdk.NewInt(8), sdk.NewInt(5000000000000),
+				v0_13hard.NewInterestRateModel(sdk.ZeroDec(), sdk.MustNewDecFromStr("0.002"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.01")),
+				sdk.MustNewDecFromStr("0.025"), sdk.MustNewDecFromStr("0.01"),
+			),
+			// usdx
+			v0_13hard.NewMoneyMarket("usdx", v0_13hard.NewBorrowLimit(true, sdk.ZeroDec(), sdk.ZeroDec()), "usdx:usd", sdk.NewInt(6), sdk.NewInt(50000000000),
+				v0_13hard.NewInterestRateModel(sdk.ZeroDec(), sdk.MustNewDecFromStr("0.002"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.01")),
+				sdk.MustNewDecFromStr("0.025"), sdk.MustNewDecFromStr("0.01"),
+			),
+			// ukava
+			v0_13hard.NewMoneyMarket("ukava", v0_13hard.NewBorrowLimit(true, sdk.ZeroDec(), sdk.MustNewDecFromStr("0.5")), "kava:usd", sdk.NewInt(6), sdk.NewInt(10000000000),
+				v0_13hard.NewInterestRateModel(sdk.ZeroDec(), sdk.MustNewDecFromStr("0.002"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.01")),
+				sdk.MustNewDecFromStr("0.025"), sdk.MustNewDecFromStr("0.01"),
+			),
+			// hard
+			v0_13hard.NewMoneyMarket("hard", v0_13hard.NewBorrowLimit(true, sdk.ZeroDec(), sdk.MustNewDecFromStr("0.5")), "hard:usd", sdk.NewInt(6), sdk.NewInt(25000000000),
+				v0_13hard.NewInterestRateModel(sdk.ZeroDec(), sdk.MustNewDecFromStr("0.002"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.01")),
+				sdk.MustNewDecFromStr("0.025"), sdk.MustNewDecFromStr("0.01"),
+			),
+		},
+		10,
+	)
+
+	for _, newDep := range v13DepositorMap {
+		v13Deposits = append(v13Deposits, newDep)
+		v13TotalSupplied = v13TotalSupplied.Add(newDep.Amount...)
+	}
+
+	for _, mm := range newParams.MoneyMarkets {
+		genAccumulationTime := v0_13hard.NewGenesisAccumulationTime(mm.Denom, GenesisTime, sdk.OneDec(), sdk.OneDec())
+		v13GenesisAccumulationTimes = append(v13GenesisAccumulationTimes, genAccumulationTime)
+	}
+
+	return v0_13hard.NewGenesisState(newParams, v13GenesisAccumulationTimes, v13Deposits, v0_13hard.DefaultBorrows, v13TotalSupplied, v0_13hard.DefaultTotalBorrowed, v0_13hard.DefaultTotalReserves)
+}
+
+// MigrateIncentive migrates from a v0.11 incentive genesis state to a v0.13 incentive genesis state
+func MigrateIncentive(hardGS v0_11hard.GenesisState, incentiveGS v0_11incentive.GenesisState) v0_13incentive.GenesisState {
+	usdxMintingRewardPeriods := v0_13incentive.RewardPeriods{}
+	usdxRewardsPerSecondMap := make(map[string]sdk.Coin)
+	usdxRewardsPerSecondMap["bnb-a"] = sdk.NewCoin("ukava", sdk.NewInt(122354))
```

Comment:
Hard Rewards expect the collateral type to match the chain denom (e.g. 'bnb') for the reward collateral type while USDX Minting expects the collateral type to match the CDP module (e.g. 'bnb-a'). Will there be any issues with using Incentive's `RewardPeriods` to back both sets of Rewards? We'd need a `RewardPeriod` for 'bnb' for Hard rewards and a 'bnb-a' `RewardPeriod` for USDX Minting rewards.

Reply:
Correct- shouldn't cause a problem although I agree there's mixing of terms (CDP's `collateral_type` and hard money markets `denom`.